src/types/device/deviceApi.ts

```diff
@@ -44,4 +52,34 @@ export default {
     TRes: Type<DeviceDetail>(),
     TBody: Type<{ location: string }>(),
   },
+
+  // Service History API
+  serviceHistory: {
+    list: {
+      path: "/api/v1/facility/{facility_external_id}/device/{device_external_id}/service_history/",
+      method: HttpMethod.GET,
```

Comment:
this is javascript; follow naming conventions; 
```suggestion
      path: "/api/v1/facility/{facilityId}/device/{deviceId}/service_history/",
      method: HttpMethod.GET,
```

Reply:
but i am passsing also like these and in api documentation these  names facility_external_id are there 
const { data: serviceHistory, isLoading } = useQuery({
    queryKey: ["device", facilityId, deviceId, "serviceHistory"],
    queryFn: query(deviceApi.serviceHistory.list, {
      pathParams: {
        facility_external_id: facilityId,
        device_external_id: deviceId,
      },
    }),

Reply:
That is on the backend. For path params, it doesn't matter what you call it so long as url path is accurate.

Reply:
i am getting 403 forbidden during calling this api can you help me?